roles/preflight-checks/tasks/neutron.yml

```diff
@@ -4,3 +4,12 @@
 
 - include: subnet.yml param={{ item }}
   with_items: "{{ neutron.subnets }}"
+
+
+- block:
+  - include: lbaas-f5.yml
+    when: neutron.lbaas.f5.enabled | bool
+
+  - include: lbaas-haproxy.yml
+    when: neutron.lbaas.f5.enabled | bool == false
+  when: (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool
```

Comment:
see previous comment on how to make when more readable using list for and conditions